.github/workflows/linters.yaml

```diff
@@ -0,0 +1,31 @@
+#
+name: linters
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+defaults:
+  run:
+    shell: bash -o pipefail {0}
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    permissions:
+      security-events: write
+      pull-requests: write
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 0
+
+      - name: Lint shell scripts
+        uses: redhat-plumbers-in-action/differential-shellcheck@latest
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          shell-scripts: ush/preamble.sh
```

Comment:
Unfortunately, this will not work as you expect. [`shell-scripts`](https://github.com/redhat-plumbers-in-action/differential-shellcheck#shell-scripts) option takes the path to a text file that contains a list of file paths that will be ignored. I know that it is a bit clumsy, and it would deserve to rework on the linter side.

But I think you could be fine by running a linter with default settings. It performs differential scans, so only new changes introduced via PR will be reported.

I'm glad you are using my action. If you have any issues or requests, feel free to reach out to me or create issue on [`differential-shellcheck`](https://github.com/redhat-plumbers-in-action/differential-shellcheck).

Reply:
Seemed to work when I tested it, but thanks for the information. It was probably just accidental that it did what I expected. I don't know that using a separate file is really all that clumsy at all.

Actually really like your action, thank you for making it available. My only request would be to make a new one that does the same for python linter ;). I'm having trouble finding one quite as well integrated into GitHub actions, to the point I am considering adapting yours for a different linter.

Reply:
I'm glad you like it. :+1: 

We are planning to create also other linters, including `differential-cppcheck` and `differential-pylint`. I can't give you any exact release date, but they are coming. They will be available at: [@redhat-plumbers-in-action](https://github.com/redhat-plumbers-in-action)